### DISCUSSION TITLE
Kanister CRDs should be distributed with helm chart

### DISCUSSION THREAD

**Discussion by viveksinghggits** [kanisterio/kanister#1232] [Ideas]
Kanister CRDs right now are distributed with kanister controller code and not the helm chart. Ideally we should try to distribute them with helm chart. 
In some scenarios people were not able to upgrade to newer version because they didn't want to give kanister access to manage CRDs and because of that upgrade to recent version failed.

**Comment by viveksinghggits** [comment]
@pavannd1 I was thinking if we should have an environment variable to figure out if the CRDs should be created using kanister controller or not. If we run kanister independently we would set that to false, otherwise (if kanister is being consumed by other projects) true.
When set to false, CRDs would be created using helm templates.

**Comment by ihcsim** [comment]
Any thoughts on how we would keep the generated vs. helm CRDs in-sync?

**Comment by viveksinghggits** [comment]
thats a good point  @ihcsim, immediate thought that I have is having symlinks between the two. We would have the main ones in helm template.
 What do you think about that?

**Comment by viveksinghggits** [comment]
I have raised a PR, closing the discussion.
https://github.com/kanisterio/kanister/pull/1244
